swarm/api/http/error_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2016 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package http
+
+import (
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/swarm/api"
+	"github.com/ethereum/go-ethereum/swarm/storage"
+)
+
+func TestError(t *testing.T) {
+	key := [3]storage.Key{}
+	dir, _ := ioutil.TempDir("", "bzz-storage-test")
+	storeparams := &storage.StoreParams{
+		ChunkDbPath:   dir,
+		DbCapacity:    5000000,
+		CacheCapacity: 5000,
+		Radius:        0,
+	}
+
+	localStore, err := storage.NewLocalStore(storage.MakeHashFunc("SHA3"), storeparams)
+	if err != nil {
+		t.Fatal(err)
+	}
+	chunker := storage.NewTreeChunker(storage.NewChunkerParams())
+	dpa := &storage.DPA{
+		Chunker:    chunker,
+		ChunkStore: localStore,
+	}
+	dpa.Start()
+	defer dpa.Stop()
+
+	maxproxyattempts := 3
+	a := api.NewApi(dpa, nil)
+	/// \todo iterate port numbers up if fail
+	StartHttpServer(a, &ServerConfig{Addr: "127.0.0.1:8569", CorsString: ""})
+	// how to wait for ListenAndServe to have initialized? This is pretty cruuuude
```

Comment:
Use [`testutil.NewTestSwarmServer`](https://github.com/ethereum/go-ethereum/blob/master/swarm/testutil/http.go#L30) (we use it in the server tests e.g. [here](https://github.com/ethereum/go-ethereum/blob/master/swarm/api/http/server_test.go#L140-L141))